app/src/main/java/com/orange/ods/app/ui/components/imageitem/ComponentImageItem.kt

```diff
@@ -0,0 +1,122 @@
+    val rightIcon = painterResource(id = R.drawable.ic_display_standard)
+    val rightIconContentDescription = stringResource(id = R.string.component_image_item_large)
+    val imageItemHeight = 250.dp
+
+    with(imageItemCustomizationState) {
+        if (!hasText) {
+            sideIcons.value = false
+        }
+        ComponentCustomizationBottomSheetScaffold(
+            bottomSheetScaffoldState = rememberBottomSheetScaffoldState(),
+            bottomSheetContent = {
+                OdsListItem(
+                    text = stringResource(id = R.string.component_element_text),
+                    trailing = OdsSwitchTrailing(checked = textDisplayed)
+                )
+                OdsListItem(
+                    text = stringResource(id = R.string.component_element_icon),
+                    trailing = OdsSwitchTrailing(checked = sideIcons, enabled = hasText)
+                )
+                Column(modifier = Modifier.padding(start = 16.dp, end = 16.dp)) {
+                    Subtitle(textRes = R.string.component_image_item_sizes)
+                    OdsSlider(
+                        value = sliderPosition,
+                        steps = 1,
+                        valueRange = 0f..20f,
+                        onValueChange = { sliderPosition = it },
+                        leftIcon = leftIcon,
+                        leftIconContentDescription = leftIconContentDescription,
+                        rightIcon = rightIcon,
+                        rightIconContentDescription = rightIconContentDescription
+                    )
+                }
+            }) {
+            Column(
+                modifier = Modifier.fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                AnimatedVisibility(
+                    visible = true, enter = fadeIn(),
+                    exit = fadeOut()
+                ) {
+                    OdsImageItem(
+                        image = rememberAsyncImagePainter(
+                            model = recipe.imageUrl,
+                            placeholder = painterResource(id = R.drawable.placeholder),
+                            error = painterResource(id = R.drawable.placeholder)
+                        ),
+                        icon = if (hasSideIcons && !iconCheckedState.value) painterResource(id = R.drawable.ic_heart_outlined)
+                        else if (hasSideIcons && iconCheckedState.value) painterResource(id = R.drawable.ic_heart) else null,
+                        title = if (hasText) recipe.title else null,
+                        modifier = if (sliderPosition.toInt() == 0) Modifier
+                            .size(174.dp, 175.dp)
+                            .padding(dimensionResource(id = R.dimen.spacing_m))
+                        else if (sliderPosition.toInt() == 20) Modifier
+                            .size(400.dp, imageItemHeight)
```

Comment:
Image width should be `fillMaxWidth()` instead of `400.dp` because you the screen width might be lower than `400.dp`.

Size for medium image should be a square:

```kotlin
size(imageItemHeight, dimensionResource(id = R.dimen.card_big_image_height))
```

Finally you can refactor this block of code using `when` because the syntax is shorter:

```kotlin
modifier = Modifier
            .run {
                when (sliderPosition.toInt()) {
                    0 -> size(175.dp, 175.dp)
                    20 -> fillMaxWidth().height(250.dp)
                    else -> size(250.dp, 250.dp)
                }
            }
            .padding(dimensionResource(id = R.dimen.spacing_m)),
```

Note: If you add a `size` property to `ImageItemCustomizationState` (see other comment), you will check `size` instead of `sliderPosition.toInt()`.